src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py

```diff
@@ -127,6 +132,33 @@ def test_azconfig_mgmt(self, resource_group, location):
 
         self.cmd('appconfig delete -n {config_store_name} -g {rg} -y')
 
+        self.cmd('appconfig show-deleted -n {config_store_name}',
+            checks=[self.check('name', '{config_store_name}'),
+                    self.check('location', '{rg_loc}'),
+                    self.check('purgeProtectionEnabled', '{enable_purge_protection}')])
+
+        deleted_stores = self.cmd('appconfig list-deleted')
+
+        found = False
+        for deleted_store in deleted_stores:
+            if deleted_store['name'] == config_store_name:
+                assert deleted_store['location'] == location
+                found = True
+        assert found
```

Comment:
There is an easier way to test this. See this for reference: https://github.com/Azure/azure-cli/blob/90e7b8b2d701168cb8850e93e4fc773eac5b46a8/src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py#L1274-L1281

Reply:
I don't think we can test this way reliably because -
1. The index at which the deleted store would be at is not fixed because any number of the stores in the same subscription can be getting created or deleted at any moment.
2. The length of deleted stores would not be fixed because of the same reason.

The reliably way to test would be to test whether the recently deleted store is present in the deleted list.

Reply:
You're right, I forgot that list-deleted works on the subscription level. In that case, should we reduce the retention period to 1 day for all tests? There's a chance that some live test might fail intermittently if a deleted store exists with the same name (even though names are random). 
We have a common method where this can be updated: `_create_config_store`, and a few other 'appconfig create' commands in some tests.

Reply:
Also, we should add a test for update command with the new --enable-purge-protection argument. Maybe update it to true and then assert that purging throws an error.
Another test would be trying to turn off purge protection after enabling it.

Reply:
> should we reduce the retention period to 1 day for all tests?
We should definitely do that.

> We should add a test for update command with the new --enable-purge-protection argument. Maybe update it to true and then assert that purging throws an error.
Another test would be trying to turn off purge protection after enabling it.

I am not sure if we should do that. These behaviors are from the server, the CLI shouldn't be testing that out. CLI should test the behavior of CLI. If the CLI had validations for these behaviors, we should have added tests for them.